Services/CO.CDP.OrganisationInformation.Persistence/DatabaseOrganisationRepository.cs

```diff
@@ -114,6 +114,41 @@ public async Task<IEnumerable<Organisation>> SearchByName(string name, PartyRole
         return await query.Select(t => t.Organisation).ToListAsync();
     }
 
+    public async Task<(IEnumerable<Organisation> Results, int TotalCount)> SearchByNameOrPpon(string name, int? limit, int skip, string orderBy)
+    {
+        var baseQuery = context.Organisations
+            .Include(b => b.Identifiers)
+            .Include(p => p.Addresses)
+            .ThenInclude(p => p.Address)
+            .Where(t =>
+                EF.Functions.ILike(t.Name, $"%{name}%") ||
+                t.Identifiers.Any(i => EF.Functions.ILike(i.IdentifierId, $"%{name}%")))
+            .Where(t => t.Type == OrganisationType.Organisation)
+            .Where(t => t.Roles.Contains(PartyRole.Buyer) || t.Roles.Contains(PartyRole.Tenderer))
+            .Where(t => t.Identifiers.Any(i => i.Scheme.Equals("GB-PPON")));
+
+        // no soring is applied if sortOrder is null or equals to rel or anything
+        if (orderBy != null)
```

Comment:
Since orderBy is not nullable, we can simplify this logic and remove unnecessary null checks